### PR TITLE
Tweak: Run hash commands sequentially. [ED-3828]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/common/assets/js/api/extras/hash-commands.js [hunk 1]
```diff
@@ -79,6 +79,7 @@ export default class HashCommands {
 	 * @param {Array.<HashCommand>} [commands=this.commands]
 	 */
 	run( commands = this.commands ) {
+		// To allow validate the method without creating promise.
```
Comment (iNewLegend) [inline core/common/assets/js/api/extras/hash-commands.js:82]: Promise should be rejected.

**Review by Nevoss — APPROVED** [review]
Nice work!

File: core/common/assets/js/api/extras/hash-commands.js [hunk 2]
```diff
@@ -78,20 +78,24 @@ export default class HashCommands {
...
 			const dispatcher = this.dispatchersList[ hashCommand.method ];
 
 			if ( ! dispatcher ) {
-				throw Error( `No dispatcher found for the command: \`${ hashCommand.command }\`.` );
+				return Promise.reject( new Error( `No dispatcher found for the command: \`${ hashCommand.command }\`.` ) );
 			}
 
 			if ( ! dispatcher.isSafe( hashCommand.command ) ) {
-				throw Error( `Attempting to run unsafe or non exist command: \`${ hashCommand.command }\`.` );
+				return Promise.reject( Error( `Attempting to run unsafe or non exist command: \`${ hashCommand.command }\`.` ) );
```
Comment (matipojo) [inline core/common/assets/js/api/extras/hash-commands.js:91]:
```suggestion
				return Promise.reject( new Error( `Attempting to run unsafe or non exist command: \`${ hashCommand.command }\`.` ) );
```

File: tests/qunit/tests/core/common/assets/js/api/extras/hash-commands.spec.js [hunk 3]
```diff
@@ -55,26 +55,57 @@ QUnit.module( 'File: core/common/assets/js/api/extras/hash-commands.js', ( hooks
...
+			method: 'e:run',
+		}, {
+			command: 'test-hash-commands/async-command',
+			method: 'e:run',
+		}, {
+			command: 'test-hash-commands/safe-command',
+			method: 'e:run',
+		} ] );
+
+		// Give him tick to reach 2.
```
Comment (matipojo) [inline tests/qunit/tests/core/common/assets/js/api/extras/hash-commands.spec.js:97]: Explain

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/extras/hash-commands.js (hunk 2 above)
Comment (iNewLegend) [reply core/common/assets/js/api/extras/hash-commands.js]: Handled.

File: tests/qunit/tests/core/common/assets/js/api/extras/hash-commands.spec.js (hunk 3 above)
Comment (iNewLegend) [reply tests/qunit/tests/core/common/assets/js/api/extras/hash-commands.spec.js:103]: Handled.

Comment (iNewLegend) [reply tests/qunit/tests/core/common/assets/js/api/extras/hash-commands.spec.js:103]: Handled.

**Review by matipojo — APPROVED** [review]

**Pull request merged**